pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -147,7 +147,7 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
       List<SegmentZKMetadata> preSelectedSegments = new ArrayList<>();
       for (SegmentZKMetadata segment : allSegments) {
         if (preSelectedSegmentsBasedOnLineage.contains(segment.getSegmentName()) && segment.getTotalDocs() > 0
-            && MergeTaskUtils.allowMerge(segment)) {
+            && MergeTaskUtils.allowMerge(segment) && segment.getStartTimeMs() != -1 && segment.getEndTimeMs() != -1) {
```

Comment:
Use `TimeUtils.timeValueInValidRange()` which is more robust

Reply:
Tried using that but a lot of tests were failing for this class. Reason being in all the tests, the timestamps used are less than `VALID_MIN_TIME_MILLIS`. I tried changing the timestamps but the relationship between them is so carefully planned so that some segments get merged while others do not and hence some tests were still failing. 
We can take this up in a different PR